plutus-core/plc/Main.hs

```diff
@@ -737,54 +737,6 @@ timeEval n evaluate prog
             pure $ (result, end - start)
 
 
----------------- Printing budgets and costs ----------------
-
-printBudgetStateBudget :: ExBudget -> IO ()
-printBudgetStateBudget b = do
-  let ExCPU cpu = _exBudgetCPU b
-      ExMemory mem = _exBudgetMemory b
-  putStrLn $ "CPU budget:    " ++ show cpu
-  putStrLn $ "Memory budget: " ++ show mem
-
-printBudgetStateTally :: (Eq fun, Hashable fun, Show fun) => Cek.CekExTally fun -> IO ()
-printBudgetStateTally (ExTally costs) = do
-  putStrLn $ "Const      " ++ pbudget Cek.BConst
-  putStrLn $ "Var        " ++ pbudget Cek.BVar
-  putStrLn $ "LamAbs     " ++ pbudget Cek.BLamAbs
-  putStrLn $ "Apply      " ++ pbudget Cek.BApply
-  putStrLn $ "Delay      " ++ pbudget Cek.BDelay
-  putStrLn $ "Force      " ++ pbudget Cek.BForce
```

Comment:
... and since we have to specify all the constraints upfront, we can't have custom printing functions downstream. A good option would be to move this code to the `Pretty` instance of `CekExTally`, but that would disrupt quite a few test cases, which is something I wanted to avoid as this PR is not tiny already. But I can restore this functionality in a subsequent PR. @kwxm what's your opinion?

Reply:
I wondered about that when I  wrote this code.  It doesn't really seem like prettyprinting, so I just stuck it in here because it's the only place it's used.  Would it be reasonable to add another type class to do this, `VerbosePrint` or something?  (I haven't look at the rest of the PR yet, so I don't know how feasible that would be.)

Reply:
I'd prefer to have a single class if possible, but otherwise yeah, we can have an additional class.

> It doesn't really seem like prettyprinting,

Why not?

Reply:
Sorry I'm late to this discussion, but maybe we want to have a way to dump it out as JSON or something and then require `ToJSON` in our bundled constraints?

Reply:
I've returned the original functionality by adding a new class for now. We'll think later on how (and if we should at all) to change that.